src/terminal/parser/stateMachine.hpp

```diff
@@ -190,7 +200,11 @@ namespace Microsoft::Console::VirtualTerminal
 
         VTIDBuilder _identifier;
         std::vector<VTParameter> _parameters;
-        bool _parameterLimitReached;
+        bool _parameterLimitOverflowed;
+        std::vector<VTParameter> _subParameters;
+        std::vector<std::pair<BYTE /*range start*/, BYTE /*range end*/>> _subParameterRanges;

```

Comment:
I've tested this PR and it works great. Thank you for your hard work! I have just one question before I approve it:
Why don't we just store the subparameter range right inside the `VTParameter` struct? That way we don't need this extra vector and the parameter slicing would be a bit simpler.

Reply:
`VTParameter` is base for both the `_parameter` and `_subParameter`. Storing subparameter range within VTParameter might make sense for `_parameter` but its should be illegal for `_subParameter`.

Reply:
I personally think it's fine to store the range for both and leave it unused for sub-parameters, because I suspect that it'll not cause serious bugs. For instance, incorrectly asking a sub-param for its sub-params in turn (= illegal) could just say that there are no sub-params available (+ `abort()` in debug builds). I'm not sure if that's better than your current approach, however. It was an idea that I had when I saw the `VTParameters::subspan` function and wondered if storing the range right inside `VTParameters` would make it simpler.

For me it's a ✅ anyways and I'll leave it up to you what you personally like best. We can still tune this code however we want in the future after all. 🙂

I'll ask @DHowett to give this just a quick look, maybe he got anything else in mind.